core/src/main/java/gyro/core/auth/UsesCredentialsDirectiveProcessor.java

```diff
@@ -29,17 +33,31 @@
     public void process(Scope scope, DirectiveNode node) {
         validateArguments(node, 1, 1);
 
+        String argument = getArgument(scope, node, String.class, 0);
+        RootScope root = scope.getRootScope();
         if (scope instanceof DiffableScope) {
             DiffableScope diffableScope = (DiffableScope) scope;
             diffableScope.getSettings(CredentialsSettings.class)
-                .setUseCredentials(getArgument(scope, node, String.class, 0));
-            if (diffableScope.getStateNodes().stream().noneMatch(o -> o.equals(node))) {
-                diffableScope.getStateNodes().add(node);
+                .setUseCredentials(argument);
+            DirectiveNode stateNode = new DirectiveNode(
+                node.getName(),
+                Collections.singletonList(new ValueNode(argument)),
+                node.getOptions(),
+                node.getBody(),
+                node.getSections());
+            if (diffableScope.getStateNodes().stream().filter(o -> o instanceof DirectiveNode).noneMatch(o -> {
+                DirectiveNode directiveNode = (DirectiveNode) o;
+                String directiveNodeArgument = (String) root.convertValue(
+                    String.class, root.getEvaluator().visit(directiveNode.getArguments().get(0), scope));
+                String stateNodeArgument = (String) root.convertValue(
+                    String.class, root.getEvaluator().visit(stateNode.getArguments().get(0), scope));
+                return directiveNode.getName().equals(stateNode.getName()) && directiveNodeArgument.equals(
+                    stateNodeArgument);
+            })) {
+                diffableScope.getStateNodes().add(stateNode);
```

Comment:
As there can only be one uses-credentials inside a block, only checking if it is present is fine I guess.

There can be a case where the variable that uses-credentials uses is changed, we still would want the state to have the old value rather than another new `@uses-credentials` entry. 

The following handles that as well.

```suggestion

            DirectiveNode stateNode = new DirectiveNode(
                node.getName(),
                Collections.singletonList(new ValueNode(argument)),
                node.getOptions(),
                node.getBody(),
                node.getSections());

            if (diffableScope.getStateNodes().stream()
                .filter(o -> o instanceof DirectiveNode)
                .map(o -> ((DirectiveNode) o))
                .noneMatch(o -> o.getName().equals(node.getName()))) {
                diffableScope.getStateNodes().add(stateNode);
```